CHANGELOG.md

```diff
@@ -21,6 +21,8 @@ The changes are relative to the previous release, unless the baseline is specifi
 * Fix CMake config shared library leaks
   https://github.com/AOMediaCodec/libavif/issues/2264.
 * Fix clang-cl compilation.
+* Report files with an item used in multiple 'dimg' boxes with
+  AVIF_RESULT_NOT_IMPLEMENTED instead of AVIF_RESULT_INVALID_IMAGE_GRID.
```

Comment:
Should we delete "instead of AVIF_RESULT_INVALID_IMAGE_GRID"? It implies that we were detecting this condition before, but the checks added in this pull request are new.

Should we also document that we now reject images that are not compliant with "within a given array, a given value shall occur at most once"?

Reply:
> Should we delete "instead of AVIF_RESULT_INVALID_IMAGE_GRID"? It implies that we were detecting this condition before, but the checks added in this pull request are new.

Such files were failing with `AVIF_RESULT_INVALID_IMAGE_GRID` before this PR, so this sentence looks accurate to me.

> Should we also document that we now reject images that are not compliant with "within a given array, a given value shall occur at most once"?

These images were already rejected with `AVIF_RESULT_INVALID_IMAGE_GRID`. So there is no change in behavior.  
I did not try to create such a file but maybe one could have created a grid with twice the same tile and another extra unused tile, so that the check `dimgItemCount == grid->rows * grid->columns` would still pass. In this very specific case, maybe a CHANGELOG entry is useful, but this is really niche.

Reply:
Did you mean these images are rejected by the following check in `avifDecoderItemReadAndParse()`?
```
            // Validate that there are exactly the same number of dimg items to form the grid.
            uint32_t dimgItemCount = 0;
            for (uint32_t i = 0; i < item->meta->items.count; ++i) {
                if (item->meta->items.item[i]->dimgForID == item->id) {
                    ++dimgItemCount;
                }
            }
            if (dimgItemCount != grid->rows * grid->columns) {
                return AVIF_RESULT_INVALID_IMAGE_GRID;
            }
```

So the new checks added in this pull request reject these images earlier (before avifDecoderItemReadAndParse()), and in one case with a different error code?

Thanks for the clarification.